rama-dns/src/chain.rs

```diff
@@ -0,0 +1,192 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use rama_net::address::Domain;
+
+use crate::DnsResolver;
+use rama_core::combinators::Either;
+
+/// An error that occurs when a DNS resolver chain fails to resolve a domain.
+#[derive(Debug)]
+pub struct DnsChainDomainResolveErr<E: 'static> {
+    errors: Vec<E>,
```

Comment:
We probably want to store `BoxError` in this `Vec<..>` and do not need a generic error type at all.

Reply:
I added Send + Sync bounds to ensure thread safe